[WOOMOB-1396] - Booking's customer section actions

AdamGrzybkowski opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->

WOOMOB-1396

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR implements the actions in the Booking details screen for both the phone and email rows in the customer details section.

There we a couple of things that I did:
1. Aligned the icons to match the Order details screen - https://github.com/woocommerce/woocommerce-android/commit/0c63f1d3a9f5456da12c2a3e84ec523c0d1ccbba
2. Extract some logic to build the list of possible actions for the phone number - https://github.com/woocommerce/woocommerce-android/commit/b16d61993dba001e35707cc76e1a14c07725f741 and https://github.com/woocommerce/woocommerce-android/commit/bb13aa454cde7a71fc2a066d08c15e7bdc0231df
3. Used that implementation on the Order screen - https://github.com/woocommerce/woocommerce-android/commit/f38810229feada2d8f73eb83ddfcc572b64257fe
4. Implemented the actions on the Booking screen - https://github.com/woocommerce/woocommerce-android/commit/8bff5bb7cfeea280446406f0912248a3077534db, https://github.com/woocommerce/woocommerce-android/commit/fc5caaa9a8fab1b441a7a97ea6f9d98719acd48b and https://github.com/woocommerce/woocommerce-android/commit/ee32e2ee2161ea95ca1c0a22f23db38c16614ca3

### Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->

1. Login and open the booking tab
2. Tap on a booking item
3. Look for `Customer` section
4. Tap on the ✉️ icon
5. Confirm an email client is open to send a new email
6. Go back to Woo app
7. Tap on the `...` icon
8. Select the options and verify 

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

You can verify the Order details screen for regression. Testing Whatsapp, Telegram is a plus. Light and dark theme.

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
The above

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->


https://github.com/user-attachments/assets/2fd2c383-349a-4033-8b36-e79b5b98e531



- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

AdamGrzybkowski committed:
Align icons for email&phone with the order screen

AdamGrzybkowski committed:
Extract sendSms function to ActivityUtils

AdamGrzybkowski committed:
Create ContactDropdownMenu composable to show contact options

AdamGrzybkowski committed:
Extract logic to check for available phone contact options

AdamGrzybkowski committed:
Use PhoneContactOptionsUtils in the Order screen

AdamGrzybkowski committed:
Use PhoneContactOptionsUtils in the Booking  screen

AdamGrzybkowski committed:
Invoke sendEmail action on contact email tap

irfano commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/bookings/compose/BookingCustomerDetails.kt:
```diff
@@ -2,36 +2,42 @@ package com.woocommerce.android.ui.bookings.compose
 import com.woocommerce.android.R
 import com.woocommerce.android.ui.compose.preview.LightDarkThemePreviews
 import com.woocommerce.android.ui.compose.theme.WooThemeWithBackground
+import com.woocommerce.android.util.ActivityUtils
 
 @Composable
 fun BookingCustomerDetails(
     model: BookingCustomerDetailsModel,
-    onEmailClick: () -> Unit,
-    onPhoneClick: () -> Unit,
     modifier: Modifier = Modifier,
 ) {
+    var phoneMenuExpanded by remember { mutableStateOf(false) }
```
> flexibility (e.g., adding analytics,

I was about to mention that too. Adding analytics would indeed be easier if we handled actions in the view models. But it’s not a major concern, we can revisit it when we implement analytics.

AdamGrzybkowski replied:
True. I think there are pros and cons, like always. This way, the component is very easy to drop and use, without needing to implement the actions every time.

irfano reviewed (approved):
Nice job, Adam! It works as expected. 👍🏻 🚀

AdamGrzybkowski committed:
Recaulculate contact options on each resume
When an app like Telegram is installed/unistalled with Woo in the background, the list will be rechecked after entering foreground and thus preventing a crash.
